src/middlewared/middlewared/plugins/reporting/netdata_web.py

```diff
@@ -0,0 +1,86 @@
+import os
+import shutil
+import threading
+import time
+
+from middlewared.schema import accepts, returns, Str
+from middlewared.service import cli_private, job, pass_app, periodic, private, CallError, Service
+from middlewared.utils import MIDDLEWARE_RUN_DIR
+from middlewared.utils.crypto import generate_string
+from passlib.apache import HtpasswdFile
+
+
+BASIC_FILE = f'{MIDDLEWARE_RUN_DIR}/netdata-basic'
+HTPASSWD_LOCK = threading.Lock()
+
+
+class ReportingService(Service):
+
+    @private
+    async def netdataweb_basic_file(self):
+        return BASIC_FILE
+
+    @cli_private
+    @accepts(roles=['READONLY_ADMIN'])
+    @returns(Str('password'))
+    @pass_app()
+    def netdataweb_generate_password(self, app):
+        """
+        Generate a password to access netdata web.
+        That password will be stored in htpasswd format for HTTP Basic access.
```

Comment:
We should note that concurrent access to netdata UI for same credentials is not supported and may lead to undesired behavior.

Reply:
Added!